File: qz/u2/santi/Exam U2/71-100/ExamSpeakers/src/main/java/ec/edu/espe/examspeakers/controller/SpeakerDAOMongoDB.java

package ec.edu.espe.examspeakers.controller;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import ec.edu.espe.examspeakers.model.Speaker;
import java.util.ArrayList;
import java.util.List;
import org.bson.Document;

/**
 *
 * @author Sjean
 */
public class SpeakerDAOMongoDB implements SpeakerDAO {
    private MongoCollection<Document> collection;

    public SpeakerDAOMongoDB() {
        String connectionString = "mongodb+srv://sjeancarlo:sjeancarlo@cluster0.3ixvnnj.mongodb.net/?appName=Cluster0";
        MongoClient mongoClient = MongoClients.create(connectionString);
        MongoDatabase database = mongoClient.getDatabase("speakerdb");
        this.collection = database.getCollection("speakers");
    }

    @Override
    public void create(Speaker speaker) {
        Document doc = new Document("_id", speaker.getId())
                .append("brand", speaker.getBrand())
                .append("power", speaker.getPower())
                .append("frequency", speaker.getFrequency());
        collection.insertOne(doc);
    }

    @Override
    public Speaker read(String id) {
        Document doc = collection.find(new Document("_id", id)).first();
        if (doc != null) {
            return new Speaker(
                doc.getString("_id"),
                doc.getString("brand"),
                doc.getDouble("power"),
                doc.getDouble("frequency")
            );
        }
        return null;
    }

    @Override
    public void update(Speaker speaker) {
        Document query = new Document("_id", speaker.getId());
        Document update = new Document("$set", new Document("brand", speaker.getBrand())
                .append("power", speaker.getPower())
                .append("frequency", speaker.getFrequency()));
        collection.updateOne(query, update);
    }

    @Override
    public void delete(String id) {
        collection.deleteOne(new Document("_id", id));
    }

    @Override
    public List<Speaker> listAll() {
        List<Speaker> speakers = new ArrayList<>();
        for (Document doc : collection.find()) {
            speakers.add(new Speaker(
                doc.getString("_id"),
                doc.getString("brand"),
                doc.getDouble("power"),
                doc.getDouble("frequency")
            ));
        }
        return speakers;
    }
}

File: qz/u3/santi/qz1&2/Quizz_Speakers/Quizz_Speakers/controller/speaker_dao_mongodb.py
from pymongo import MongoClient
from model.speaker import Speaker
from controller.speaker_dao import SpeakerDAO
from typing import List

class SpeakerDAOMongoDB(SpeakerDAO):
    def __init__(self):
        connection_string = "mongodb+srv://sjeancarlo:sjeancarlo@cluster0.3ixvnnj.mongodb.net/?appName=Cluster0"
        client = MongoClient(connection_string)
        database = client["speakerdb"]
        self.collection = database["speakers"]

    def create(self, speaker: Speaker):
        doc = {
            "_id": speaker.id,
            "brand": speaker.brand,
            "power": speaker.power,
            "frequency": speaker.frequency
        }
        self.collection.insert_one(doc)

    def read(self, speaker_id: str) -> Speaker:
        doc = self.collection.find_one({"_id": speaker_id})
        if doc:
            return Speaker(
                doc["_id"],
                doc["brand"],
                doc["power"],
                doc["frequency"]
            )
        return None

    def update(self, speaker: Speaker):
        query = {"_id": speaker.id}
        update = {
            "$set": {
                "brand": speaker.brand,
                "power": speaker.power,
                "frequency": speaker.frequency
            }
        }
        self.collection.update_one(query, update)

    def delete(self, speaker_id: str):
        self.collection.delete_one({"_id": speaker_id})

    def list_all(self) -> List[Speaker]:
        speakers = []
        for doc in self.collection.find():
            speakers.append(Speaker(
                doc["_id"],
                doc["brand"],
                doc["power"],
                doc["frequency"]
            ))
        return speakers
